core/domain/summary_services.py

```diff
@@ -256,3 +256,19 @@ def get_top_rated_exploration_summary_dicts(language_codes):
         reverse=True)[:NUMBER_OF_TOP_RATED_EXPLORATIONS]
 
     return _get_displayable_exp_summary_dicts(sorted_exp_summaries)
+
+
+def get_recently_published_exploration_summary_dicts(language_codes):
+    """Returns a list of recently published explorations
+     with the given language code.
+    """
+    collect_all_exp_summaries = [
+        exp_summary for exp_summary in
```

Comment:
Interesting. So, if one of the recently-published explorations is not in English, you end up showing < 8 explorations? Seems incorrect...

Three possibiilities:
- Just show all recently-published explorations regardless of language code.
- Amend the datastore query so that it includes a language code equality check.
- Fetch a bit more than 8 explorations, so that after we drop some, we'll (probably) still have more than 8.

Thoughts? @kevinlee12?